src/proxy_handler.cpp

```diff
@@ -0,0 +1,206 @@
+//
+// Created by Michael Chen on 2/28/17.
+// Copyright © 2017 Chen. All rights reserved.
+//
+//ProxyHandler code for generating new request and capturing response is modeled off of
+//http://www.boost.org/doc/libs/1_55_0/doc/html/boost_asio/example/cpp03/http/client/sync_client.cpp
+
+#include "proxy_handler.h"
+#include "parsed_config.h"
+#include <istream>
+#include <ostream>
+#include <string>
+#include <vector>
+#include <boost/bind.hpp>
+#include <boost/asio.hpp>
+#include "utils.h"
+#include <boost/regex.hpp>
+
+
+RequestHandler::Status ProxyHandler::Init(const std::string& uri_prefix,
```

Comment:
Nice job with this function - using existing ParsedConfig class to extract values. 👍